diff --git a/node_modules/expo-dev-launcher/android/build.gradle b/node_modules/expo-dev-launcher/android/build.gradle
--- a/node_modules/expo-dev-launcher/android/build.gradle
+++ b/node_modules/expo-dev-launcher/android/build.gradle
@@ -130,6 +130,6 @@ dependencies {
   debugOnly project(":expo-dev-menu-interface")
   debugOnly project(":expo-manifests")
   debugOnly project(":expo-updates-interface")
-    androidTestApi project(":expo-dev-menu")
+  debugOnly project(":expo-dev-menu")
 
   debugOnly 'commons-io:commons-io:2.6'
