diff --git a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewManagerDefinition.kt b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewManagerDefinition.kt
--- a/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewManagerDefinition.kt
+++ b/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/views/ViewManagerDefinition.kt
@@ -38,6 +38,9 @@ class ViewManagerDefinition(
     val nativeModulesProxy = reactContext.getUnimoduleProxy() ?: return
     val appContext = nativeModulesProxy.kotlinInteropModuleRegistry.appContext
 
-    appContext.jsLogger.error(exception)
+    appContext.jsLogger?.error(
+      message = exception.message.toString(),
+      cause = exception
+    )
   }
 }
diff --git a/node_modules/expo-modules-core/android/build.gradle b/node_modules/expo-modules-core/android/build.gradle
--- a/node_modules/expo-modules-core/android/build.gradle
+++ b/node_modules/expo-modules-core/android/build.gradle
@@ -207,5 +207,4 @@ android {
 }
 
 dependencies {
-  implementation 'com.android.support:animated-vector-drawable:28.0.0'
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:2.1.20"
